src/App.tsx

```diff
@@ -1,61 +1,119 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+const findUserById = (id: number): User => {
+  return usersFromServer.filter(user => user.id === id)[0];
+};
+
+const initialTodos: Todo[] = todosFromServer.map((todo) => {
+  return {
+    id: todo.id,
+    title: todo.title,
+    completed: todo.completed,
+    user: findUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [titleTodo, setTitleTodo] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [todos, setTodos] = useState(initialTodos);
+
+  const [emptyTitle, setEmptyTitle] = useState(false);
+  const [isSelectedUser, setIsSelectedUser] = useState(false);
+
+  const addTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (titleTodo === '') {
+      setEmptyTitle(true);
+    }
+
+    if (selectedUser === 0) {
+      setIsSelectedUser(true);
+    }
+
+    if (selectedUser === 0 || titleTodo === '') {
+      return;
+    }
+
+    const todo: Todo = {
+      id: todos.length > 0
+        ? (todos.sort((t1, t2) => t2.id - t1.id)[0].id) + 1
+        : 0,
+      completed: false,
+      title: titleTodo,
+      user: findUserById(selectedUser),
+    };
```

Comment:
Don't change your original 'todo', your code work incorrect 
Better make copy of it